README.md

```diff
@@ -3,59 +3,332 @@
+
+* [Way-Too-Quick Start](#way-too-quick-start)
+* [Installation](#installation)
+  * [Build From Source With Cargo](#build-from-source-with-cargo)
+  * [Install Prebuilt Binaries With Cargo-Binstall](#install-prebuilt-binaries-with-cargo-binstall)
+  * [Download From Github Releases](#download-from-github-releases)
+* [Setup](#setup)
+  * [Configuring Installers](#configuring-installers)
+  * [Configuring Targets](#configuring-targets)
+* [Usage (CI)](#usage-ci)
+* [Usage (Local Builds)](#usage-local-builds)
+* [Concepts](#concepts)
+* [Build Flags](#build-flags)
+* [Compatibility With Other Tools](#compatibility-with-other-tools)
+* [Contributing](#contributing)
+  * [Updating Snapshots](#updating-snapshots)
+  * [Cutting Releases](#cutting-releases)
+
+
+
+# Way-Too-Quick Start
+
+```sh
+# install tools
+cargo install cargo-dist
+cargo install cargo-release
+
+# one-time setup
+cargo dist init --ci=github
+
+# cut a release
+cargo release 0.1.0
+```
+
+That's gonna do a whole bunch of stuff you might not have expected (but I left off the --execute flag from `cargo-release` so you won't actually break anything if you really did just copy paste that 😇). Read the rest of the docs to know what it does!
+
+
+
+
+
+# Installation
+
+## Build From Source With Cargo
+
+```sh
+cargo install dist --profile=dist
+```
+
+## Install Prebuilt Binaries With Cargo-Binstall
+
+```sh
+cargo binstall cargo-dist
+```
+
+## Download From Github Releases
+
+[Latest Release](https://github.com/axodotdev/cargo-dist/releases/latest)
+
+
+
+
+
+# Setup
+
+Once cargo-dist is installed, you can set it up in your cargo project by running
+
+```sh
+cargo dist init --ci=github
+```
+
+This will:
+
+* Add a `dist` build profile to your Cargo.toml (with recommended default build flags)
+* Add a `.github/workflows/release.yml` file to your project (only if you pass `--ci=...`)
+
+These changes should be checked in to your repo for whenever you want to cut a release.
+
+If you don't want ci scripting generated, but just want the `dist` profile you can do:
+
+```sh
+cargo dist init
+```
+
+If you want to just (re)generate the ci scripts, you can do:
+
+```sh
+cargo dist generate-ci github
+```
+
+See the next section ("Usage (CI)") for how the github workflow is triggered and what it does.
+
+
+
+## Configuring Installers
+
+If you would like to generate (currently bad) installer scripts, you can pass `--installer` flags
+to either `init` or `generate-ci`:
+
+```sh
+cargo dist init --ci=github --installer=github-shell --installer=github-powershell
+```
+
+This will result in `installer.sh` and `installer.ps1` being generated which fetch from a Github Release™️ and copy the binaries to `HOME/.cargo/bin/` on the assumption that this is on your path. The scripts are currently brittle and won't properly tell cargo about the installation. As such they are currently only really appropriate for bootstrapping CI scripts. This will be improved in the future.
```

Comment:
we may want to try to caveat this for folks who may not care if cargo knows about the installation e.g. people who build rust tools for other ecosystems..

Reply:
Hmm could you elaborate?

Reply:
thinking about this more, it's not clear if they're useful for that *either* because then `.cargo/bin` is unlikely to be on the PATH.

Reply:
oh dang right, hrm. yeah ok. you're right here.